accounts/auth_tools/auth_middleware.py

```diff
@@ -23,8 +19,14 @@ def __call__(self, request):
             response = self.get_response(request)
         except User.DoesNotExist:
             request.user = None
-            response = self.get_response(request)            
-        except (jwt.exceptions.DecodeError, jwt.exceptions.ExpiredSignatureError, json.decoder.JSONDecodeError):
-            response = HttpResponse()
-            response.status_code = 401
+            response = self.get_response(request)
+        except jwt.exceptions.DecodeError:
+            request.user = None
+            response = self.get_response(request)
+        except jwt.exceptions.ExpiredSignatureError:
+            request.user = None
+            response = self.get_response(request)
+        except json.decoder.JSONDecodeError:
+            request.user = None
+            response = self.get_response(request)
```

Comment:
Why are you using multiple except blocks to the same behaviour instead of catching multiple exceptions at the same time?

Is there any advantages or different behaviour expected for each one of those?

```suggestion
        except (jwt.exceptions.DecodeError,
            jwt.exceptions.ExpiredSignatureError,
            json.decoder.JSONDecodeError):
            request.user = None
            response = self.get_response(request)
```